web/modules/custom/custom_multistep_form/src/Plugin/Block/ConfigureFormStepsBlock.php

```diff
@@ -12,13 +12,22 @@
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
- * Provides a 'FormConfigureBlock' block.
+ * Provides a 'ConfigureFormStepsBlock' block.
  */
 #[Block(
-  id: "form_configure_block",
-  admin_label: new TranslatableMarkup("Configure Form Steps"),
+  id: "configure_form_steps_block",
+  admin_label: new TranslatableMarkup("Configure Multistep Form Steps"),
 )]
-class FormConfigureBlock extends BlockBase implements ContainerFactoryPluginInterface {
+class ConfigureFormStepsBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Steps names.
+   */
+  protected $stepNames = [
+    1 => 'Products',
+    2 => 'Delivery',
+    3 => 'Payment',
+  ];
```

Comment:
One of the main things was that there is no point in associating the steps with the meaningless numbers, yet you are still doing that.
It is awful to see half-ass solutions/fixes, unless it is agreed to do that way due to time restrictions/simplicity whatever.
You SHOULD think and pay MUCH more attention to how your code reads to the others, that is one of the key aspects of your work